### DISCUSSION TITLE
Central Authentication Service (CAS)

### DISCUSSION THREAD

**Discussion by dotmobo** [open-webui/open-webui#7631] [Ideas]
Hi,

I am currently working on a [Central Authentication Service (CAS)](https://en.wikipedia.org/wiki/Central_Authentication_Service) integration.  
It is an alternative to the [OAuth](https://en.wikipedia.org/wiki/OAuth) protocol.

This authentication system is widely used in some universities.

I have implemented an initial working version of CAS authentication using [python-cas](https://github.com/python-cas/python-cas).

Would you be interested in a pull request for this?

Regards,  
Morgan

**Comment by matthew-kusz** [comment]
We are looking at using open webui at our institution and need the option to use CAS. What is the status of this?

**Comment by lijiext** [comment]
I need this feature for internal using.

**Comment by dotmobo** [comment]
After some research, you can use the following environment variables when running the app to work with CAS:  

```
OPENID_REDIRECT_URI=https://youropenwebuiurl/oauth/oidc/callback
OPENID_PROVIDER_URL=https://yourcasurl/cas/oidc/.well-known/openid-configuration
```  

This will enable OpenID for CAS: https://apereo.github.io/cas/7.0.x/authentication/OIDC-Authentication.html
